registry/remote/example_test.go

```diff
@@ -0,0 +1,424 @@
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	digest := "sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec"
+	descriptor, err := repo.Resolve(ctx, digest) // Still need to resolve first, don't create a new descriptor with the digest, blob size is unknown
+	if err != nil {
+		panic(err) // Handle error
+	}
+	r, err := repo.Fetch(ctx, descriptor) // Fetch the blob from the repository
+	if err != nil {
+		panic(err) // Handle error
+	}
+	defer r.Close() // don't forget to close
+
+	// option 1: fetch a WHOLE blob
+	pulledBlob, err := io.ReadAll(r)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(string(pulledBlob))
+
+	// option 2: partially read, if the remote registry supports
+	if seeker, ok := r.(io.ReadSeeker); ok {
+		_, err = seeker.Seek(8, io.SeekStart)
+		if err != nil {
+			panic(err) // Handle error
+		}
+		pulledBlob, err := io.ReadAll(r)
+		if err != nil {
+			panic(err) // Handle error
+		}
+		fmt.Println(string(pulledBlob))
+	}
+
+	// caller SHOULD verify what is fetched
+	if descriptor.Digest != ocidigest.FromBytes(pulledBlob) || descriptor.Size != int64(len(pulledBlob)) {
+		panic(err) // Handle error
+	}
+
+	// Output:
+	// Example blob content
+	// blob content
+}
+
+// ExampleRepository_Tag gives example snippets for tagging a descriptor.
+func ExampleRepository_Tag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	// suppose we are going to tag a blob with below digest
+	digest := "sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec"
+
+	// 1. Resolve the target desc
+	descriptor, err := repo.Resolve(ctx, digest)
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	// 2. Tag the resolved desc
+	tag := "latest"
+	err = repo.Tag(ctx, descriptor, tag)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println("Succeed")
+
+	// Output:
+	// Succeed
+}
+
+// ExampleRegistry_Repositories gives example snippets for listing respositories in a HTTPS registry with pagination.
+func ExampleRegistry_Repositories() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	// Override the `host` variable to play with local registry.
+	// Uncomment below line to reset HTTP option:
+	// reg.PlainHTTP = true
+	ctx := context.Background()
+	err = reg.Repositories(ctx, func(repos []string) error {
+		for _, repo := range repos {
+			fmt.Println(repo)
+		}
+		return nil
+	})
+	if err != nil {
+		panic(err) // Handle error
+	}
+	// Output:
+	// public/repo1
+	// public/repo2
+	// internal/repo3
+}
+
+func Example_pullByTag() {
```

Comment:
What is the different between this and the `ExampleRepository_Fetch_byTag()`?  Just the use of `Fetch()` and `FetchAll()`?
The name makes me feels like this function is to "pull" the whole image, or do something different than "fetch"...

Reply:
Since the `FetchAll` belongs to the package content, we can only use scenario suffix `pullByXXX` here